internal/appsec/appsec.go

```diff
@@ -25,78 +26,78 @@ func Enabled() bool {
 	return activeAppSec != nil && activeAppSec.started
 }
 
+func newConfigWithOpts(opts ...options.StartOption) (*config.Config, error) {
+	cfg, err := config.NewConfig()
+	if err != nil {
+		return nil, err
+	}
+	for _, opt := range opts {
+		opt(cfg)
+	}
+
+	return cfg, nil
+}
+
 // Start AppSec when enabled is enabled by both using the appsec build tag and
 // setting the environment variable DD_APPSEC_ENABLED to true.
-func Start(opts ...config.StartOption) {
+func Start(opts ...options.StartOption) {
+	cfg, err := newConfigWithOpts(opts...)
+	if err != nil {
+		logUnexpectedStartError(err)
+		return
+	}
+
 	telemetry := newAppsecTelemetry()
 	defer telemetry.emit()
 
 	// AppSec can start either:
 	// 1. Manually thanks to DD_APPSEC_ENABLED
-	// 2. Remotely when DD_APPSEC_ENABLED is undefined
-	// Note: DD_APPSEC_ENABLED=false takes precedence over remote configuration
+	// 2. Manually thanks to options.WithCodeActivation(true)
+	// 2. Remotely when DD_APPSEC_ENABLED and options.WithCodeActivation are undefined
+	// Note: DD_APPSEC_ENABLED=false and options.WithCodeActivation(false) takes precedence over remote configuration
 	// and enforces to have AppSec disabled.
-	enabled, set, err := config.IsEnabled()
```

Comment:
I haven't read the full logic, but here is what's important:  `DD_APPSEC_ENABLED=false` must always disable AppSec, no matter what else is trying to enable it. So code activation + DD_APPSEC_ENABLED=false => appsec disabled.